### ISSUE TITLE
Case insensitive sorting?

### ISSUE THREAD

**Issue by david-pfx** [louischatriot/nedb#580]
Is there any way to do case insensitive sorting? Any way at all?

I note Issue 140, but that as in 2014. Any progress?

Just a simple callback on two sort keys would solve the problem.

**Comment by beaugunderson** [comment]
@david-pfx i just ran into this need as well--i think a sort callback would be a great way to solve this but since i don't think that's going to happen i'm going to solve for my project by denormalizing a document property just for sorting, e.g. saving the `toLowerCase()` version in a separate property

**Comment by beaugunderson** [comment]
oh, this is actually the suggestion from #150 as well (creating sort keys)

**Comment by beaugunderson** [comment]
there is also the `compareStrings` option to `Datastore`, but it's global to the `Datastore` (that may be fine for your use case, though!)

> **compareStrings** (optional): function compareStrings(a, b) compares strings a and b and return -1, 0 or 1. If specified, it overrides default string comparison which is not well adapted to non-US characters in particular accented letters. Native localCompare will most of the time be the right choice

**Comment by beaugunderson** [comment]
and for completeness' sake, an example:

```js
    compareStrings: (a, b) => {
      return a.toLowerCase().localeCompare(b.toLowerCase());
    },
```

**Comment by david-pfx** [comment]
Yes, that does seem to be the state of play. One of these days I think I'm going to have to fork NeDB, but other things are taking priority right now.

**Comment by JamesMGreene** [comment]
MongoDB doesn't even support this functionality for sorting AFAIK, so I can't see NeDB adding it any time soon.

https://docs.mongodb.com/manual/reference/method/cursor.sort/

**Comment by polomoshnov** [comment]
NeDB seems to not use `compareStrings` when comparison operators such as `$lt` and `$gt` are used.

Here's an example:
```js
(async function () {
    const db = new Nedb({
      compareStrings: (a, b) => {
        return a.toLowerCase().localeCompare(b.toLowerCase());
      },
    });
  
    for (const title of ['A1', 'B', 'C', 'a2']) {
      await new Promise(resolve => db.insert({ title }, resolve));
    }

    function find(query) {
      return new Promise(resolve => db.find(query).sort({ title: 1 }).projection({ _id: 0 }).exec((err, res) => resolve(res)));
    }

    console.log(JSON.stringify({
      find: await find({}),
      find2: await find({ title: { $lt: 'a2' } }),
    }, null, 2));
  })();
```

The outcome is as follows:
```js
{
  "find": [
    {
      "title": "A1"
    },
    {
      "title": "a2"
    },
    {
      "title": "B"
    },
    {
      "title": "C"
    }
  ],
  "find2": [
    {
      "title": "A1"
    },
    {
      "title": "B"
    },
    {
      "title": "C"
    }
  ]
}
```

If `compareStrings` worked with comparison operators then `find2` would contain only `A1`. But judging from the outcome we can conclude that the lowercase title goes after the uppercase ones irrespective of the `compareStrings` option.
